src/fabric_cicd/publish.py

```diff
@@ -63,71 +76,111 @@ def publish_all_items(fabric_workspace_obj: FabricWorkspace, item_name_exclude_r
         )
         fabric_workspace_obj.publish_item_name_exclude_regex = item_name_exclude_regex
 
+    # Collect published items information
+    published_items = {}
+    
+    # Helper function to add items to the published_items dictionary
+    def collect_published_items_for_type(item_type: str) -> None:
```

Comment:
Can you suggest an alternate place to have this function, maybe a separate file or outside of this function?